### PR TITLE
planner: set EnableOuterJoinReorder to false by default (#37264)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #37264 to release-6.1
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/37280
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/37280:release-6.1-21847fe58b51
```

---

### What problem does this PR solve?

Issue Number: ref #37238 

Problem Summary:

### What is changed and how it works?

Set it to false by default.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by fixdb — APPROVED** [review]
+1

**Review by chrysan — APPROVED** [review]

File: sessionctx/variable/sysvar_test.go [hunk 1]
```diff
@@ -677,6 +677,68 @@ func TestSettersandGetters(t *testing.T) {
...
+				TiDBOptTiFlashConcurrencyFactor,
+				TiDBOptSeekFactor:
+				continue
+			}
+			require.Equal(t, false, sv.skipInit, fmt.Sprintf("skipInit should not be set on new system variables. variable %s is in violation", sv.Name))
+		}
+	}
+}
+
+>>>>>>> 21847fe58... planner: set EnableOuterJoinReorder to false by default (#37264)
```
Comment (fixdb) [inline sessionctx/variable/sysvar_test.go:741]: @winoros please fix this

**Review by AilinKid — APPROVED** [review]

**Pull request merged**